File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.py
# 대진 라운드인가 검사
# a b 비교 
# 주의 -> abs(a - b) = 1 인가? -> 반례 a = 2, b = 3  
# 추가 -> 둘 중 큰 수가 짝수인가 검사
def IsOnRound(a,b):
    return (a - b == 1 or a - b == -1) and max(a,b) % 2 == 0

def solution(n,a,b):                   
    round = 1    
    
    while not IsOnRound(a,b):
        # 다음 라운드의 번호 알아오기
        # 1. num = 홀수인 경우 + 1
        # 2. num / 2
        # 3. round + 1
        if(a%2==1): a+= 1
        a //= 2
        
        if(b%2==1): b+= 1
        b //= 2
                
        round += 1
        print(a, b, round)

    return round

File: 프로그래머스/2/12985. 예상 대진표/예상 대진표.cpp
#include <iostream>
using namespace std;
//대진 라운드인가 검사 //a b 비교 
//주의 -> abs(a - b) = 1 인가? -> 반례 a = 2, b = 3  
//추가 -> 둘 중 큰 수가 짝수인가 검사
bool IsOnRound(int a, int b)
{
    bool check1 = (a - b == 1 || a - b == -1);
    
    bool check2;
    if (a > b) check2 = (a % 2 == 0);
    else check2 = (b % 2 == 0);

    cout << a << ' ' << b << endl;
    cout << check1 << check2 << endl;
    return check1 && check2;
}      

int solution(int n, int a, int b)
{
    int answer = 1;
    while(!IsOnRound(a,b)) 
    {
        if(a%2==1) a+= 1;
        a /= 2;

        if(b%2==1) b+= 1;
        b /= 2;

        answer++;        
    }             
    return answer;
}